### PR TITLE
Feature/layout - Initial layout added

### PR DESCRIPTION
**Pull request by asuvarnaeworldes79**
Features added:
 1. Header
2. Footer
3. SIdebar
4. Layout 

Note: style follows global theme

imageimage image

### CODE REVIEW

File: src/components/Layout/components/Header/components/RightNavigation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import Box from '@mui/material/Box';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import * as Icons from '@mui/icons-material';
+
+type IconName = keyof typeof Icons;
+
+interface RightNavigationItem {
+    label: string;
```
Comment (anilkumarganhari) [inline src/components/Layout/components/Header/components/RightNavigation/index.tsx]: What if the navigation item has a child navigation items?

File: src/components/Layout/components/Header/components/UserMenu/index.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+              {data.avatar || data.username.charAt(0).toUpperCase()}
+            </Avatar>
+          </IconButton>
+          <Typography sx={{ textAlign: 'center', fontSize: '0.875rem' }}>
+            {data.username}
+          </Typography>
+        </Box>
+      </Tooltip>
+      <Menu
+        sx={{ mt: '45px' }}
```
Comment (anilkumarganhari) [inline src/components/Layout/components/Header/components/UserMenu/index.tsx]: no px use rem

File: src/components/Layout/components/Header/components/RightNavigation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import Box from '@mui/material/Box';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import * as Icons from '@mui/icons-material';
+
+type IconName = keyof typeof Icons;
+
+interface RightNavigationItem {
```
Comment (anilkumarganhari) [inline src/components/Layout/components/Header/components/RightNavigation/index.tsx]: createa seaprate file for interfaces

File: src/components/Layout/components/Header/components/RightNavigation/index.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    label: string;
+    icon?: string;
+    path?: string;
+    handler?: () => void;
+}
+
+interface RightNavigationProps {
```
Comment (anilkumarganhari) [inline src/components/Layout/components/Header/components/RightNavigation/index.tsx]: move to seperate file

File: src/components/Layout/components/Header/components/UserMenu/index.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import Menu from '@mui/material/Menu';
+import Avatar from '@mui/material/Avatar';
+import Tooltip from '@mui/material/Tooltip';
+import MenuItem from '@mui/material/MenuItem';
+
+interface UserMenuProps {
```
Comment (anilkumarganhari) [inline src/components/Layout/components/Header/components/UserMenu/index.tsx]: All interfaces should be into a separate file and export then index.ts

File: src/components/Layout/components/Header/components/UserMenu/index.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorElUser(event.currentTarget);
+  };
+
+  const handleCloseUserMenu = () => {
+    setAnchorElUser(null);
+  };
+
+  return (
+    <Box sx={{ flexGrow: 0, borderLeft: '1px solid #ddd', pl: 2 }}>
```
Comment (anilkumarganhari) [inline src/components/Layout/components/Header/components/UserMenu/index.tsx]: use rem, no px

File: package-lock.json
Comment (anilkumarganhari) [inline package-lock.json:1]: do not check-in the package-lock.json

File: src/components/Layout/components/Footer/index.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        <Box
+            component="footer"
+            sx={{
+                py: 2,
+                textAlign: "center",
+                backgroundColor: "rgba(0, 0, 0, 0.05)",
+            }}
+        >
+            <Typography variant="body2" color="text.secondary">
+                Footer Placeholder
```
Comment (anilkumarganhari) [inline src/components/Layout/components/Footer/index.tsx]: what is this place holder. how the conenet will be displayed here wihout props?

File: src/components/Layout/components/Header/data/index.tsx
Comment (anilkumarganhari) [inline src/components/Layout/components/Header/data/index.tsx:1]: this should as part of mock json,  not part of index.ts

File: src/components/Layout/components/PageHeader/index.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+} from "@mui/material";
+import { PageHeaderProps } from '../../../../types/page-header';
+
+
+const PageHeader: React.FC<PageHeaderProps> = ({ title, secondaryTitle, infoText, infoSecondaryText, chipStyles }: PageHeaderProps) => {
+    return (
+        <Box sx={{ display: "flex", flexDirection: "column", p: 2, backgroundColor: "rgba(0, 0, 0, 0.05)", }}>
+            <Breadcrumbs aria-label="breadcrumb">
+                <Link underline="hover" color="inherit" href="#">
+                    Home
```
Comment (anilkumarganhari) [inline src/components/Layout/components/PageHeader/index.tsx]: no harded code text, should come from props

File: src/components/Layout/components/PageHeader/index.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+                </Link>
+                <Typography color="text.primary">Child Support Services</Typography>
```
Comment (anilkumarganhari) [inline src/components/Layout/components/PageHeader/index.tsx]: no harded code text, should come from props

File: package-lock.json
Comment (anilkumarganhari) [inline package-lock.json:1]: remove this lock.josn file. never include in your check-in

File: src/components/Layout/components/Header/components/RightNavigation/index.tsx (hunk 1 above)
Comment (asuvarnaeworldes79) [reply src/components/Layout/components/Header/components/RightNavigation/index.tsx]: This Right Navigation is an example component. Header comp will accept any valid React Node

Comment (anilkumarganhari) [reply src/components/Layout/components/Header/components/RightNavigation/index.tsx]: what if the Route as child routes, does it handle?

**Review by anilkumarganhari — APPROVED** [review]

**Comment by anilkumarganhari** [comment]
Closing this PR....

**Pull request closed without merging**